cJSON.c

```diff
@@ -882,10 +889,22 @@ char *cJSON_PrintBuffered(const cJSON *item, int prebuffer, cjbool fmt)
     }
     p.length = prebuffer;
     p.offset = 0;
+    p.noalloc = 0;
 
     return print_value(item, 0, fmt, &p);
 }
 
+int cJSON_PrintPreallocated(cJSON *item,char *buf, const size_t len, const cjbool fmt)
+{
+    char *out;
+    printbuffer p;
+    p.buffer = buf;
+    p.length = len;
+    p.offset = 0;
+    p.noalloc = 1;
+    out = print_value(item,0,fmt,&p);
+    return (out != buf ? -1 : 0);
```

Comment:
If this function returns a char pointer (effectively providing an alias to the input `char *buf` or a NULL pointer), it might falsely imply a new dynamically allocated string was created. Do you think it might be more intuitive to use the pointer to `char* buf` after the function is called if no error was returned, or is consistency of return values with all the `cJSON_Print` functions more important?